__device-tests__/utils.js

```diff
@@ -51,17 +51,29 @@ export async function takeScreenshotByElement( element, { padding } = {} ) {
  *
  * @typedef {Object} CropScreenshot
  *
- * @property {number}         x                         X offset, in pixels.
- * @property {number}         y                         Y offset, in pixels.
- * @property {number}         width                     Width, in pixels.
- * @property {number}         height                    Height, in pixels.
+ * @property {number}               x                         X offset, in pixels.
+ * @property {number}               y                         Y offset, in pixels.
+ * @property {number}               width                     Width, in pixels.
+ * @property {number}               height                    Height, in pixels.
  *
- * @param    {Object}         options                   Options
- * @param    {boolean}        [options.withoutKeyboard] Prevents showing the keyboard in the screenshot.
- * @param    {CropScreenshot} [options.crop]            Specify values to crop the screenshot.
+ * @typedef {Object} CropOffsetScreenshot
+ *
+ * @property {number}               left                      Left offset to apply to the crop values, in pixels.
+ * @property {number}               top                       Top offset to apply to the crop values, in pixels.
+ * @property {number}               right                     Right offset to apply to the crop values, in pixels.
+ * @property {number}               bottom                    Bottom offset to apply to the crop values, in pixels.
```

Comment:
Maybe a crazy idea: is there a parameter design that allows us to avoid adding a new `cropOffset` parameter?

I'm wonder how we might use negative numbers to achieve the outcome we desire without expanding the API surface area. 

```js
takeScreenshot({  right: -20  });
```

The issue with the current design — which is not directly used anywhere outside of `takeScreenshotByElement` — is that it does not have a `right` value, but uses a combination of `x` and `width`. Would it be worthwhile to redesign these arguments to enable negative values? Would we be able to retain the simplicity in use of, say, `takeScreenshot({  withoutKeyboard: true  })`? 

Off the top of my head, I am picturing something like the following. 

```js
@typedef {Object} CropScreenshot

@property {number} top    Top offset, in pixels.
@property {number} right  Right offset, in pixels.
@property {number} bottom Bottom offset, in pixels.
@property {number} left   Left offset, in pixels.
@property {number} width  Width, in pixels.
@property {number} height Height, in pixels.
```

Is this worthwhile or a waste of time?

Reply:
I thought about adding the `right` and `bottom` offset parameters to the `crop` object, but I felt that it could be confusing to have two different coordinate formats in the same object (i.e. `x, y` format vs `top, left` format). My rationale for creating a second parameter was mainly driven by that fact.

> I'm wonder how we might use negative numbers to achieve the outcome we desire without expanding the API surface area.
> 
> ```js
> takeScreenshot({  right: -20  });
> ```

Ah, interesting. After checking your comment, I realized that using the word `right` here might be confusing. The way I see the offset values to transform the surface area is more like the [`padding` CSS property](https://developer.mozilla.org/en-US/docs/Web/CSS/padding). So if you set a right value, it would push the content to the left. This makes me think that probably the word `offset` can be misleading for this parameter. We could change it to `padding` to better match what we try to achieve by cutting some pixels from the borders. WDYT?

> Off the top of my head, I am picturing something like the following.
> 
> ```js
> /**
>  * @typedef {Object} CropScreenshot
>  *
>  * @property {number} top    Top offset, in pixels.
>  * @property {number} right  Right offset, in pixels.
>  * @property {number} bottom Bottom offset, in pixels.
>  * @property {number} left   Left offset, in pixels.
>  * @property {number} width  Width, in pixels.
>  * @property {number} height Height, in pixels.
>  */
> ```
> 
> Is this worthwhile or a waste of time?

Yeah, this approach is interesting. My only concern is how the width and height affect the other parameters. Could you share some examples? It would help me to better understand how each parameter interacts with the cropping surface area. Thanks!

Alternatively, and following my comment regarding the padding, we could have the following parameters:

```js
/**
 * @typedef {Object} CropSurface
 *
 * @property {number}         x                         X offset, in pixels.
 * @property {number}         y                         Y offset, in pixels.
 * @property {number}         width                     Width, in pixels.
 * @property {number}         height                    Height, in pixels.
 *
 * @typedef {Object} CropPadding
 *
 * @property {number}         left                      Padding to apply to the left side, in pixels.
 * @property {number}         top                       Padding to apply to the top side, in pixels.
 * @property {number}         right                     Padding to apply to the right side, in pixels.
 * @property {number}         bottom                    Padding to apply to the bottom side, in pixels.
 *
 * @typedef {Object} CropScreenshot
 *
 * @property {CropSurface}    surface                   Surface dimensions to crop the screenshot. The default values will try to fit the screenshot size.
 * @property {CropPadding}    padding                   Padding to apply to the cropping surface. By default, no padding is applied.
 */
```

WDYT?

Reply:
As I reflect on my thoughts further, I believe they may miss my original goal of simplifying the utility. 

I do not believe using the name "padding" would clarify things either, as (1) `padding` is already used in `takeScreenshotByElement` for the inverse purpose — adding space — and (2) negative padding isn't a concept in CSS. 

My recommendation would be to leave the implementation as-is or remove it and use a hard-coded `width` value instead, e.g. `takeScreenshot({ width: WIDTH_SANS_SCROLLBAR })`. The latter could make sense as the device properties are static and this is (currently) a one-off usage. 

WDYT?

Reply:
> My recommendation would be to leave the implementation as-is or remove it and use a hard-coded `width` value instead, e.g. `takeScreenshot({ width: WIDTH_SANS_SCROLLBAR })`. The latter could make sense as the device properties are static and this is (currently) a one-off usage.

I'm hesitant to use a hard-coded value for the sake of keeping the tests as device-agnostic as possible. I agree we currently use a single device, but if in the future we use a different one or even multiple, this approach would need to be revisited.

One option I'm thinking of is to use `takeScreenshotByElement` and target the block list element. This way we'll capture the entire content area and using negative padding should crop the screenshots as we need. WDYT?

Reply:
Fair enough regarding device-agnostic tests. 

Using the approach `takeScreenshotByElement` and negative padding sounds like it could work well. Do you think we'd overload the `padding` argument to accept both a number (`-20`) and object (`{ right: -20 }`) or require an object with top/right/bottom/left? I think I lean towards overloading so that we retain the easy of use for universal padding with a number.

Reply:
> Yep, I agree with overloading the argument and accepting both number and object 👍.

Done https://github.com/wordpress-mobile/gutenberg-mobile/pull/6135/commits/9bd3294236378186ac4e428361da874cfaacd514 and used https://github.com/wordpress-mobile/gutenberg-mobile/pull/6135/commits/9a12cfac8e7f25ffab99ef0d8c4f2d787160fbe2.